src/xdr-types/bool.js

```diff
@@ -0,0 +1,104 @@
+import * as dom from 'dts-dom';
+
+/**
+ * Builds a valid dts-dom node representing a XDR Bool.
+ *
+ */
+export default function xdrBool(ns) {
+  const buffer = dom.create.interface('Buffer');
+  const boolInterface = dom.create.interface('Bool');
+
+  boolInterface.members.push(
+    dom.create.method(
+      'read',
+      [dom.create.parameter('io', buffer)],
+      dom.type.boolean
+    )
+  );
+  boolInterface.members.push(
+    dom.create.method(
+      'write',
+      [
+        dom.create.parameter('value', dom.type.boolean),
+        dom.create.parameter('io', buffer)
+      ],
+      dom.type.void
+    )
+  );
+  boolInterface.members.push(
+    dom.create.method(
+      'isValid',
+      [dom.create.parameter('value', dom.type.boolean)],
+      dom.type.boolean
+    )
+  );
+  boolInterface.members.push(
+    dom.create.method(
+      'toXDR',
+       [dom.create.parameter('value', dom.type.boolean)],
+        buffer
+      )
+  );
+  boolInterface.members.push(
+    dom.create.method(
+      'fromXDR',
+      [
+        dom.create.parameter('input', buffer),
+        dom.create.parameter(
+          'format',
+          dom.type.stringLiteral('raw'),
+          dom.ParameterFlags.Optional
+        )
+      ],
+      dom.type.boolean
+    )
+  );
+  boolInterface.members.push(
+    dom.create.method(
+      'fromXDR',
+      [
+        dom.create.parameter('input', dom.type.string),
+        dom.create.parameter(
+          'format',
+          dom.create.union([
+            dom.type.stringLiteral('hex'),
+            dom.type.stringLiteral('base64')
+          ])
+        )
+      ],
+      dom.type.boolean
+    )
+  );
```

Comment:
It always felt silly to me that we have 

```typescript
    fromXDR(input: Buffer, format?: 'raw'): Buffer;
    fromXDR(input: string, format: 'hex' | 'base64'): Buffer;
```
when we could consolidate into
```typescript
    fromXDR(input: string, format: 'raw' | 'hex' | 'base64'): Buffer;
```
but this aligns with the other generators so :man_shrugging: